builder/processor/src/main/java/io/helidon/builder/processor/TypeHandler.java

```diff
@@ -111,37 +117,31 @@ TypeName builderGetterType(boolean required, boolean hasDefault) {
 
     String generateBuilderGetter(boolean required, boolean hasDefault) {
         if (builderGetterOptional(required, hasDefault)) {
-            return "java.util.Optional.ofNullable(" + name + ")";
+            return ClassModel.TYPE_TOKEN + "java.util.Optional" + ClassModel.TYPE_TOKEN + ".ofNullable(" + name + ")";
         } else {
             return name;
         }
     }
 
-    String fieldDeclaration(PrototypeProperty.ConfiguredOption configured,
-                            boolean isBuilder,
-                            boolean alwaysFinal) {
-        StringBuilder fieldDeclaration = new StringBuilder("private ");
-
-        if (alwaysFinal || !isBuilder) {
-            fieldDeclaration.append("final ");
-        }
+    Field.Builder fieldDeclaration(PrototypeProperty.ConfiguredOption configured,
```

Comment:
I think this should get a parameter of ClassModel and create the field on it (as that would be consistent with other changes that are part of this PR).
Or return a builder, and then change other methods to return builders of their parts as well (I would not prefer this option)

Reply:
Since we need to modify created field builder in case of polymorphism, we have decided with Tomas to keep this as is.